src/code/z_kankyo.c

```diff
@@ -557,38 +583,39 @@ f32 Environment_LerpWeightAccelDecel(u16 endFrame, u16 startFrame, u16 curFrame,
     return ret;
 }
 
-void func_8006FB94(EnvironmentContext* envCtx, u8 unused) {
-    if (envCtx->gloomySkyMode != 0) {
-        switch (envCtx->unk_DE) {
-            case 0:
-                if ((envCtx->gloomySkyMode == 1) && !gSkyboxBlendingEnabled) {
-                    envCtx->unk_19 = 1;
-                    envCtx->unk_17 = 0;
-                    envCtx->unk_18 = 1;
-                    envCtx->unk_1A = 100;
-                    envCtx->unk_21 = 1;
-                    envCtx->unk_1F = 0;
-                    envCtx->unk_20 = 2;
-                    D_8011FB34 = 2;
-                    envCtx->unk_22 = envCtx->unk_24 = 100;
-                    envCtx->unk_DE++;
+void Environment_UpdateStorm(EnvironmentContext* envCtx, u8 unused) {
+    if (envCtx->stormRequest != STORM_REQUEST_NONE) {
+        switch (envCtx->stormState) {
+            case STORM_STATE_OFF:
+                if ((envCtx->stormRequest == STORM_REQUEST_START) && !gSkyboxIsChanging) {
+                    envCtx->changeSkyState = CHANGE_SKY_REQUESTED;
+                    envCtx->skyboxConfig = 0;
+                    envCtx->nextSkyboxConfig = 1;
+                    envCtx->changeSkyTimer = 100;
```

Comment:
this is getting complicated, there are two kinds of "change" things

`gSkyboxIsChanging` refers to changing between skybox indices (for an entry inside the config);
the other `change*` things are refering to a config change

This could be addressed by adding config to change* stuff and `gSkyboxIsChanging` -> `gSkyboxIndexIsChanging` I guess

Reply:
imo `gSkyboxIsChanging` is appropriate as is, it signifies the skybox itself is currently changing instead of the config. with the others now mentioning that the config is whats changing I think its clear that its separate

Reply:
Well with a config change the skybox itself changes too right?
Both changing config and changing setting is done by lerping to the next thing (and changing config checks `gSkyboxIsChanging` is false to make sure both don't change at the same time)

So since the effect in game sounds similar (fading to a different skybox) I figured this could use clarification

I'm fine with this now that I understand the code, but if I was to discover it again, I wouldn't rule out `gSkyboxIsChanging` being about the config change being active

At the very least `gSkyboxIsChanging` may deserve a comment like
```c
// Indicates whether the skybox is changing to a different index of the same config (based on time)
u8 gSkyboxIsChanging = false;
```
idk not sure about my wording there. Don't want to explain the whole thing either